src/legacy_wrappers/cursors.js

```diff
@@ -0,0 +1,187 @@
+'use strict';
+
+const { maybeCallback } = require('../utils');
+const { toLegacy } = require('../utils');
+
+module.exports = Object.create(null);
+Object.defineProperty(module.exports, '__esModule', { value: true });
+
+const commonCursorFunctions = new Map([
```

Comment:
There are only four methods in this class - is this work abstracting out into a map and then iteratively building the prototype?  It's extra obfuscation for minimal benefit.  

If we do want to reduce code duplication, then deleting the map and having free functions for each common cursor function would be my preference.  Then we could just manually set the functions onto the prototype of each cursor - just 6 lines of code - and no loop.  That seems much simpler.